src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,310 @@
+     * @param amt Gem amount.
+     */
+    function push(uint256 amt) external isMate {
+        _doPush(amt);
+    }
+
+    /**
+     * @notice Flushes out any GEM balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external isMate {
+        _doQuit(gem.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out specific amount of GEM balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param amt Gem amount.
+     */
+    function quit(uint256 amt) external isMate {
+        _doQuit(amt);
+    }
+
+    /**
+     * @notice Flushes out specific `amt` of `token` to `usr` address.
+     * @dev Can only be called by the admin
+     * @param token Token address.
+     * @param usr Destination address.
+     * @param amt Token amount.
+     */
+    function yank(
+        address token,
+        address usr,
+        uint256 amt
+    ) external auth {
+        GemAbstract(token).transfer(usr, amt);
+        emit Yank(token, usr, amt);
+    }
+
+    /**
+     * @notice Calculate required amount of GEM to get `wad` amount of DAI.
+     * @param wad DAI amount.
+     * @return gemAmt Amount of GEM required.
+     */
+    function requiredGemAmt(uint256 wad) external view returns (uint256 gemAmt) {
+        return mul(wad, WAD) / mul(sub(WAD, psm.tin()), to18ConvertionFactor);
+    }
+
+    /**
+     * @notice Swaps the specified amount of GEM into DAI through the PSM and push it into the `to` address.
+     * @param amt GEM amount.
+     */
+    function _doPush(uint256 amt) internal {
+        require(to != address(0), "RwaInputConduit3/invalid-to-address");
+
+        psm.sellGem(to, amt);
+
+        emit Push(to, _gemToDai(amt));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of GEM to the `quitTo` address.
+     * @param amt GEM amount.
+     */
+    function _doQuit(uint256 amt) internal {
+        require(quitTo != address(0), "RwaInputConduit3/invalid-quit-to-address");
+        gem.transfer(quitTo, amt);
+        emit Quit(quitTo, amt);
+    }
+
+    /**
+     * @notice DAI received for swapping `gemAmt` of GEM
+     * @param amt GEM amount.
+     */
+    function _gemToDai(uint256 amt) internal view returns (uint256) {
+        uint256 amt18 = mul(amt, to18ConvertionFactor);
+        uint256 fee = mul(amt18, psm.tin()) / WAD;
+        return sub(amt18, fee);
+    }
+
+    /*//////////////////////////////////
+                    Math
+    //////////////////////////////////*/
+
+    uint256 internal constant WAD = 10**18;
+
+    function sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, "Math/sub-overflow");
+    }
+
+    function mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "Math/mul-overflow");
+    }
```

Comment:
Consider adding `_` prefix to silent warnings.

Reply:
done